### PR TITLE
fix(listview): account for panel orientation

### LINKED ISSUE #13196: Keyboard navigation in ListView should account for orientation
### Current behavior

Currently, when pressing up and down, the ListView moved to previous and next items. If the ItemPanelTemplate is a Panel that has a horizontal orientation, the same thing still happens.

### Expected behavior

Navigation should use left and right arrow keys instead if the orientation is horizontal.

### How to reproduce it (as minimally and precisely as possible)

```xaml
<ListView ItemsSource="123456789qweasdzxc"
	        ScrollViewer.HorizontalScrollBarVisibility="Visible"
	        ScrollViewer.HorizontalScrollMode="Enabled"
	        ScrollViewer.VerticalScrollMode="Disabled">
	<ListView.ItemsPanel>
		<ItemsPanelTemplate>
			<ItemsStackPanel Orientation="Horizontal" />
		</ItemsPanelTemplate>
	</ListView.ItemsPanel>
</ListView>
```

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #13196

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 4a44ab9</samp>

This pull request adds an `InternalOrientation` property to the `Panel` class and its subclasses that have an orientation, such as `StackPanel`, `ItemsStackPanel`, and `ItemsWrapGrid`. This property is used by the `ListViewBase` class to improve keyboard navigation for horizontal and boundary cases.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs [hunk 1]
```diff
@@ -105,13 +105,21 @@ internal bool TryHandleKeyDown(KeyRoutedEventArgs args)
...
 			}
-			else if (args.Key == VirtualKey.Down)
-			{
-				return TryMoveKeyboardFocusAndSelection(+1, args.KeyboardModifiers);
-			}
-			else if (args.Key == VirtualKey.Up)
+			else
 			{
-				return TryMoveKeyboardFocusAndSelection(-1, args.KeyboardModifiers);
+				var orientation = ItemsPanelRoot.InternalOrientation ?? Orientation.Vertical;
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:110]: nit: InternalOrientation can be non-nullable with default implementation returning Vertical.

Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs:110]: It might be good to assume that `ItemsPanelRoot` could be null.

**Review by Youssef1313 — COMMENTED** [review]
Why is the `InternalOrientation` not overridden in `VariableSizedWrapGrid`?

And maybe also `WrapGrid` (though it's not implemented)

**Review by Youssef1313 — COMMENTED** [review]
`CalendarPanel` has an internal orientation property. Not sure if it should override `InternalOrientation` as well.

**Comment by ramezgerges** [comment]
As far as I've tested, ListView only works with StackPanel, ItemsWrapGrid, and ItemsStackPanel. On Uno, ItemsWrapGrid doesn't seem to be implemented fully, so it's not working. I might override InternalOrientation for the remaining types for completeness.

**Comment by Youssef1313** [comment]
> As far as I've tested, ListView only works with StackPanel, ItemsWrapGrid, and ItemsStackPanel.

Is this the same as in WinUI?

**Comment by Youssef1313** [comment]
> I might override InternalOrientation for the remaining types for completeness.

It might be good to do so, assuming other parts of the code used this property in the future 😉

**Comment by ramezgerges** [comment]
> Is this the same as in WinUI?

I'm talking about WinUI, yes.

**Comment by jeromelaban** [comment]
@ramezgerges we may be able to add tests for this, though I'm not sure if the keyboard interfactions on the Uno.UITest allow for sending such keys. If not possible, let's add a new Sample Page with the manual test attribute  and textual explanations of what to test, so we can have it validated again in the future.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**